reference-architecture/gce-ansible/playbooks/openshift-setup.yaml

```diff
@@ -0,0 +1,50 @@
+- include: /usr/share/ansible/openshift-ansible/playbooks/byo/config.yml
+  vars:
+    debug_level: 5
+    console_port: 443
+    openshift_debug_level: "{{ debug_level }}"
+    openshift_node_debug_level: "{{ node_debug_level | default(debug_level, true) }}"
+    osm_controller_args:
+      cloud-provider:
+      - "gce"
+      cloud-config:
+      - "/etc/origin/gce.ini"
+    osm_api_server_args:
+      cloud-provider:
+      - "gce"
+      cloud-config:
+      - "/etc/origin/gce.ini"
+    openshift_node_kubelet_args:
+      cloud-provider:
+      - "gce"
+      cloud-config:
+      - "/etc/origin/gce.ini"
+      node-labels:
+      - "role={{ openshift_node_labels.role }}"
+    openshift_master_debug_level: "{{ master_debug_level | default(debug_level, true) }}"
+    openshift_master_access_token_max_seconds: 2419200
+    openshift_master_api_port: "{{ console_port }}"
+    openshift_master_console_port: "{{ console_port }}"
+    osm_cluster_network_cidr: 172.16.0.0/16
+    openshift_registry_selector: "role=infra"
+    openshift_router_selector: "role=infra"
+    openshift_hosted_router_replicas: 2
+    openshift_hosted_registry_replicas: 2
+    openshift_master_cluster_method: native
+    openshift_master_cluster_hostname: "internal-openshift-master.{{ public_hosted_zone }}"
+    openshift_master_cluster_public_hostname: "openshift-master.{{ public_hosted_zone }}"
+    osm_default_subdomain: "{{ wildcard_zone }}"
+    osm_default_node_selector: "role=app"
+    openshift_deployment_type: openshift-enterprise
+    openshift_master_identity_providers:
```

Comment:
We should be using the Google oAuth provider here:

    deployment_type: openshift-enterprise
    openshift_master_identity_providers:
    - name: google
      kind: GoogleIdentityProvider
      login: true
      challenge: false
      mapping_method: claim
      clientID: XXXXXXXXXX-YYYYYYYYYYYYYYYYYYYYYYYYYYYY.apps.googleusercontent.com
      clientSecret: ZZZZZZZZZZZZZZZZZZZZZZZZZZ
      hostedDomain: "" (Optional) [1]

[1] Optional hosted domain to restrict sign-in accounts to. If empty, any Google account is allowed to authenticate.